src/lock_file/satisfiability.rs

```diff
@@ -307,6 +307,18 @@ pub fn verify_pypi_platform_satisfiability(
             ));
         };
 
+        // Error if a conda package with same name is provided.
+        if environment
```

Comment:
This results in a never satisfiable lockfile:
```
 WARN pixi::lock_file::pypi: pandoc exists in both Conda and PyPI dependencies
 WARN pixi::lock_file::pypi: pandoc exists in both Conda and PyPI dependencies
 WARN pixi::lock_file::pypi: pandoc exists in both Conda and PyPI dependencies
 WARN pixi::lock_file::pypi: pandoc exists in both Conda and PyPI dependencies
```

Reply:
Can you elaborate what you mean? In my case I'm getting only one warning, not sure why you got multiple 🤔

Reply:
I test on pixi, which is multiplatform so i guess it gives the warning per platform. But I the resulting warning is just to show it always happens. 

I expect no warning and I expect my lock file to be satisfied.

Reply:
I think the warning comes from the check above (`pypi.rs`). In that case, the lock file is already satisfied but it shows warning. Isn't that the expected behavior?

Reply:
```toml
[dependencies]
pandoc = "*"

[pypi-dependencies]
pandoc = "*"
```
This is a correct toml as it these packages are different in the different eco-systems.

But because of this check it will never satisfy the lockfile.